src/aolayer.cpp

```diff
@@ -498,7 +506,6 @@ void AOLayer::movie_ticker()
   }
   //  qint64 difference = elapsed - movie_delays[frame];
   if (frame >= movie_frames.size()) {
-    m_reader.jumpToImage(frame);
     movie_frames.resize(frame + 1);
     movie_frames[frame] = this->get_pixmap(m_reader.read());
     movie_delays.resize(frame + 1);
```

Comment:
Maybe I didn't catch this before, but `resize` is an expensive operation that can cause reallocation.

Reply:
Switched to using .append() as of https://github.com/AttorneyOnline/AO2-Client/pull/302/commits/dd69d015327ba4a33b422c3d09f005455347a85f.

Per the Qt 5 documentation, .append() is the same as calling .resize(size() + 1), but QVectors are optimized with this in mind.